API.md

```diff
@@ -0,0 +1,106 @@
+## Models
+
+### Blueprint
+TBD
+
+### Header
+
+```
+{
+	type:      'header';
+	id:        number;
+	sourceUrl: string; // URL of the original page from which the header was created.
+	// More fields TBD.
+}
+```
+### Footer
+
+```
+{
+	type:      'footer';
+	id:        number;  
+	sourceUrl: string; // URL of the original page from which the footer was created.
+	// More fields TBD.
+}
+```
```

Comment:
The more I think about header, footer and navigation in general, the less things are clear for me :)

If we consider that it's acceptable for the user to modify the header/footer directly through WordPress, my suggestion would be to double-down on the KISS principle you mention, and don't deal with header/footer at all for now. The user will need to go into WordPress to author header/footer, for now.

Instead, we could first tackle Pages, because very likely the navigation will reference Pages, so having those done before tackling navigation would make sense to me.

We can also consider to never handle navigation. If we consider it to be acceptable for the user to author navigation directly in WordPress, having try-wordpress import navigation will be of limited usefulness.

Reply:
I agree for the most part, but I think we should handle navigation at the bare minimum. Considering it's a universal thing, every source site is going to have it, our WP theme in PG preview would have it, so seems like a low-hanging fruit to make the end result look good.

The only thing that's icky about is, it doesn't fit our current mental model of subjects, but considering every subject has its type, we can easily handle every subject type differently (this being the exception as its not pure data, but rather meta data about the site).

As far as handling pages go, I think handling simple text pages is quite simple with what we have already built (exactly as posts), but think of other kind of pages such as landing pages with different kind of sections making up the page. Here, I think we would be building some kind of parser to recognize stack of sections that make up the page and transform them to blocks and/or patterns to really liberate them, otherwise all context is gone.